### DISCUSSION TITLE
How can I embed a child image or a child canvas as an image inside the current text note?

### DISCUSSION THREAD

**Discussion by capi** [TriliumNext/Trilium#2967] [Q&A] [answered]
I may be missing something, but if I drag an image to a note, it is embedded (and added as a child to the current page). If I remove the image, I now see the image as a child page (otherwise it is hidden, since it is embedded. I like that a lot!)

How can I re-embed the image in the text-note? The "Insert Image" from the toolbar allows to upload a new image, but that's not what I want. Similar the "Include Note" does not seem to be the right thing here.

Similar, if I create a canvas child-note, how can I add it as an image to the current text note?

**Comment by zadam** [comment, marked as answer]
Hi, if I understand correctly, these are 2 questions:

> How can I re-embed the image in the text-note? The "Insert Image" from the toolbar allows to upload a new image, but that's not what I want.

Use the "copy to clipboard" button, it copies the reference to the image which you can then paste in a text note:

![image](https://user-images.githubusercontent.com/617641/178066556-68eb5e29-dddd-408f-a1a6-2a2c50aa06be.png)

> Similar, if I create a canvas child-note, how can I add it as an image to the current text note?

You can use "include note" to have it visible in the text note. But there's no hiding of canvas note in this case, that applies only to images.
